### PR TITLE
feat: add posthog

### PR DESCRIPTION
**Pull request by owlas**
Features:
- don't identify opt-out users
- events are anonymous before signin/register
- anonymous events before sign-in are attributed to user after sign-in
- new session begins with new anonymous user after sign-out
- sensitive components masked
- users are assigned to groups
- use react library for posthog provider
- use react library for feature flags

Also see issue: https://github.com/lightdash/lightdash-internal-work/issues/961

### CODE REVIEW

File: packages/frontend/src/pages/SqlRunner.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ import { SectionName } from '../types/Events';
 
 const generateBasicSqlQuery = (table: string) =>
     `SELECT *
-   FROM ${table} LIMIT 25`;
+     FROM ${table} LIMIT 25`;
```
Comment (rephus) [inline packages/frontend/src/pages/SqlRunner.tsx:49]: unrelated change

**Review by rephus — APPROVED** [review]
Code looks good, I haven't tested it though

**Comment by owlas** [comment]
rebasing
